UnrealPlugin/README.md

```diff
@@ -105,7 +105,116 @@ public class <projectname>ServerTarget : TargetRules
+**"Instructions for No Existing GameInstance Class"**.
+
+Locate your GameInstance class, which is most likely called something similar to [your game name]GameInstance. From now on, your game instance class will
+be denoted with [YourGameInstanceClassName].
+
+First, check the include statements and ensure that the following are included:
+
+```cpp
+#include "CoreMinimal.h"
+#include "Engine/GameInstance.h"
+#include "MyGameInstance.generated.h"
+```
+
+In the header file for your GameInstance class ([YourGameInstanceClassName].h), add the following declarations to the public section:
+(If you already have an Init() function, do not include this)
+
+```cpp
+public:
+
+    virtual void Init() override;
+    virtual void OnStart() override;
+```
+
+Then, add the following declarations to the protected section of methods:
+```cpp
+protected:
+
+    UFUNCTION()
+    void OnGSDKShutdown();
+    
+    UFUNCTION()
+    bool OnGSDKHealthCheck();
+};
+```
+
+Then, locate your [YourGameInstanceClassName].cpp file.
+
+Make sure that the following are included:
+
+```cpp 
+#include "[YourGameInstanceClassName].h"
+#include "PlayfabGSDK.h"
+#include "GSDKUtils.h"
+```
+
+Then, check in [YourGameInstanceClassName].cpp file to see if you have a boolean or variable that indicates whether the instance is 
+for a dedicated server. **If you can find this variable**, then add in this in at the end of your Init() function:
+
+```cpp
+	if (IsDedicatedServerInstance() == true)
+	{
+		FOnGSDKShutdown_Dyn OnGsdkShutdown;
+		OnGsdkShutdown.BindDynamic(this, &UShooterGameInstance::OnGSDKShutdown);
+		FOnGSDKHealthCheck_Dyn OnGsdkHealthCheck;
+		OnGsdkHealthCheck.BindDynamic(this, &UShooterGameInstance::OnGSDKHealthCheck);
+
+		UGSDKUtils::RegisterGSDKShutdownDelegate(OnGsdkShutdown);
+		UGSDKUtils::RegisterGSDKHealthCheckDelegate(OnGsdkHealthCheck);
+
+		OnStart();
+	}
+```
+**If you can't find a variable like IsDedicatedServerInstance(),** we still want to make sure that ReadyForPlayers() and onStart() are used when using 
+a dedicated server, so you could wrap the call to ReadyForPlayers() as such at the bottom of the Init function:
+
+```cpp
+#if UE_SERVER
+    if (!UGSDKUtils::ReadyForPlayers())
+    {
+        FPlatformMisc::RequestExit(false);
+    }
+#endif
+```
+
+And add these method implementations to the bottom of [YourGameInstanceClassName].cpp:
+
+```cpp
+void UMyGameInstance::OnStart()
+{
+    UE_LOG(LogTemp, Warning, TEXT("Reached onStart!"));
+    if (!UGSDKUtils::ReadyForPlayers())
+    {
+        FPlatformMisc::RequestExit(false);
+    }
+}
+
+void UMyGameInstance::OnGSDKShutdown()
+{
+    UE_LOG(LogTemp, Warning, TEXT("Shutdown!"));
+    FPlatformMisc::RequestExit(false);
+}
+
+bool UMyGameInstance::OnGSDKHealthCheck()
+{
+    UE_LOG(LogTemp, Warning, TEXT("Healthy!"));
+    return true;
+}
+```
+
+### Instructions for No Existing GameInstance Class
```

Comment:
I think I would put this above at line 109. "If you don't have a GameInstance, create one with the instructions in this step. If you already have one, skip to the next step". This would allow the instructions would be sequential, rather than jumping forward and back.

Reply:
OH, I see more how it gets to this. Yeah, this would require some re-writing, but I think these two sections could be merged into a single flow. Instead of creating and pasting the whole files in the "create" section, we could instruct them to create the default files (if missing), and then move on to the "modify" section (swap places so modify is the next section), to add the individual functions and such. The final code clips would still contain whole files, but described as an example file, rather than one they should copy paste whole. That would involve merging and de-dupe-ing these two sections.

Reply:
Thank you for this feedback! I've made some updates that are along the lines of making it a single flow instead of Existing vs New.